frontend/public/components/utils/request-size-input.tsx

```diff
@@ -19,12 +21,14 @@ export const RequestSizeInput: React.FC<RequestSizeInputProps> = ({
   required,
   step,
   testID,
+  defaultSize,
+  numberSpinner,
 }) => {
   const [unit, setUnit] = React.useState(defaultRequestSizeUnit);
-  const [value, setValue] = React.useState(defaultRequestSizeValue);
+  const [value, setValue] = React.useState(_.toInteger(defaultRequestSizeValue));
 
   const onValueChange: React.ReactEventHandler<HTMLInputElement> = (event) => {
-    setValue(event.currentTarget.value);
+    setValue(_.toInteger(event.currentTarget.value));
```

Comment:
```suggestion
    setValue(event.currentTarget.value);
```
Make the conversion outside of the generic component unless you want to change how all the users of this component handle this value.